src/one_dragon_qt/windows/main_app_window_base.py

```
@@ -109,6 +152,96 @@ def _show_context_notify(self, event: ContextNotifyEvent) -> None:
+
+    def _emit_download_request(self, event: ContextEventItem) -> None:
+        """将下载确认请求通过信号传递到主线程。"""
+        if isinstance(event.data, ContextDownloadRequestEvent):
+            self.context_download_request_signal.emit(event.data)
+
+    def _show_download_request_dialog(self, event: ContextDownloadRequestEvent) -> None:
+        """将后端资源下载请求交给框架协调器聚合处理。"""
+        self.resource_update_coordinator.add_context_download_request(event)
+
+    def closeEvent(self, event: QCloseEvent) -> None:
+        """活动下载存在时先请求异步取消，避免强行结束线程。"""
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)